rpc/tests/rpc_command_tests/get_blocks_in_chain.rs

```diff
@@ -18,5 +19,65 @@ async fn get_blocks_in_chain_ok() {
 
 fn generate_get_blocks_command_for_main_chain() -> GetBlocksInChain {
     let chain_id = get_main_chain_id_by_tag();
-    GetBlocksInChain { chain_id }
+    GetBlocksInChain::with_default_params(chain_id)
+}
+
+#[tokio::test]
+async fn get_blocks_with_optional_params_bad_head_hash_fail() {
+    let bad_head_hash = "".to_string();
+    let command = generate_get_blocks_command_with_explicit_params(None, Some(bad_head_hash), None);
+    let client = get_rpc_client();
+    assert!(client.check_node_online().await);
+
+    let client_response = client.execute(&command).await;
+    assert!(client_response.is_err());
+}
+
+#[tokio::test]
+async fn get_blocks_from_head_optional_params_ok() {
+    let head_hash = "BKmXPAiniarmJAxvv3T3CQ9sa42TbLiDdE6c57Gc74NjcKNFQBE".to_string();
+    let command = generate_get_blocks_command_with_explicit_params(None, Some(head_hash), None);
+
+    let client = get_rpc_client();
+    assert!(client.check_node_online().await);
+
+    let client_response = client.execute(&command).await;
+    assert!(client_response.is_ok());
+
+    let block_response = client_response.unwrap();
+    let blocks = block_response.block_ids.into_vec();
+    assert!(blocks.len() >= 1);
+}
+
+#[tokio::test]
+async fn get_blocks_optional_length_ok() {
+    let length = 5;
+    let command = generate_get_blocks_command_with_explicit_params(Some(length), None, None);
+
+    let client = get_rpc_client();
+    assert!(client.check_node_online().await);
+
+    let client_response = client.execute(&command).await;
+    assert!(client_response.is_ok());
+
+    let block_response = client_response.unwrap();
+    let mut blocks = block_response.block_ids.into_vec();
+
+    let last_block_batch = blocks.pop();
+    assert!(last_block_batch.is_some());
+
+    assert_eq!(last_block_batch.unwrap().len(), length as usize);
+}
```

Comment:
These query tests are mainly checking if the response from the RPC operates how we expect it to but it's not really testing if the query we attached to the command URL is correct. Adding some more tests, or modifying these, that explicitly tests the query string would verify if `GetBlocksInChainParameters` is working correctly and it's something that we might want. lmk what you guys think

Reply:
Completely agree, I don't think adding the tests in this module is correct, but some unit tests are definitely overdue.

Reply:
With that being said, it's not worth it to be exhaustive, since we trust that `querystring` crate has done their work and that the stringified query works well enough.

Reply:
Yeah that's a good point!